### ISSUE TITLE
Crash when adding a custom node on a behavior tree.

### ISSUE THREAD

**Issue by MineBill** [FlaxEngine/FlaxEngine#1586]
**Issue description:**
Crash when adding a custom node on a behavior tree.

**Steps to reproduce:**
I copied the custom node code from [the docs](https://github.com/FlaxEngine/FlaxDocs/blob/588b16e9bfd23018bce133eb368754086a25a959/manual/scripting/ai/behavior-trees/custom-node.md#custom-delay-node) and tried adding it to a behavior tree. The editor then crashes with the following message:
![image](https://github.com/FlaxEngine/FlaxEngine/assets/30367251/984b8a5d-9b05-418a-b69d-4e9e4e818c88)
On another project that i'm using to learn about behavior trees i get the same crash when pressing play.
The crash doesn't produce any logs.

**Minimal reproduction project:**

[MyProject.zip](https://github.com/FlaxEngine/FlaxEngine/files/12786216/MyProject.zip)

**Flax version:**
Flax 1.7 dd21020297ba40c338ecc84293405ce02816110b

**Comment by MineBill** [comment]
After some digging, it looks like this issue has to do with calling `GetDebugInfo` on the custom node.
Here is the stacktrace i get after attaching the debugger:
```
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at System.Runtime.CompilerServices.Unsafe.Read[[System.IntPtr, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](Void*)
   at FlaxEngine.Interop.NativeInterop+MarshalHelper`1+ReferenceTypeField`1[[FlaxEngine.BehaviorUpdateContext, FlaxEngine.CSharp, Version=1.7.6400.0, Culture=neutral, PublicKeyToken=null],[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].ToManagedFieldValueType(System.Reflection.FieldInfo, Int32, FlaxEngine.BehaviorUpdateContext ByRef, IntPtr, Int32 ByRef)
   at FlaxEngine.Interop.NativeInterop+MarshalHelperValueType`1[[FlaxEngine.BehaviorUpdateContext, FlaxEngine.CSharp, Version=1.7.6400.0, Culture=neutral, PublicKeyToken=null]].ToManagedWithMarshallableFields(FlaxEngine.BehaviorUpdateContext ByRef, IntPtr, Boolean)
   at FlaxEngine.Interop.NativeInterop+MarshalHelper`1[[FlaxEngine.BehaviorUpdateContext, FlaxEngine.CSharp, Version=1.7.6400.0, Culture=neutral, PublicKeyToken=null]].ToManaged(FlaxEngine.BehaviorUpdateContext ByRef, IntPtr, Boolean)
   at FlaxEngine.Interop.NativeInterop+MarshalHelper`1[[FlaxEngine.BehaviorUpdateContext, FlaxEngine.CSharp, Version=1.7.6400.0, Culture=neutral, PublicKeyToken=null]].ToManagedUnbox(IntPtr)
   at FlaxEngine.Interop.NativeInterop+Invoker+InvokerRet1`3[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[FlaxEngine.BehaviorUpdateContext, FlaxEngine.CSharp, Version=1.7.6400.0, Culture=neutral, PublicKeyToken=null]].InvokeThunk(System.Object, FlaxEngine.Interop.ManagedHandle, IntPtr*)
   at FlaxEngine.Interop.NativeInterop+ThunkContext.InvokeThunk(FlaxEngine.Interop.ManagedHandle, IntPtr, IntPtr, IntPtr, IntPtr, IntPtr, IntPtr, IntPtr)
   at FlaxEngine.Behavior.<Internal_GetNodeDebugInfo>g____PInvoke|43_0(IntPtr, IntPtr)
   at FlaxEngine.Behavior.<Internal_GetNodeDebugInfo>g____PInvoke|43_0(IntPtr, IntPtr)
   at FlaxEngine.Behavior.Internal_GetNodeDebugInfo(IntPtr, IntPtr)
   at FlaxEngine.Behavior.GetNodeDebugInfo(FlaxEngine.BehaviorTreeNode, FlaxEngine.Behavior)
   at FlaxEditor.Surface.Archetypes.BehaviorTree+NodeBase.UpdateDebugInfo(FlaxEngine.BehaviorTreeNode, FlaxEngine.Behavior)
   at FlaxEditor.Surface.Archetypes.BehaviorTree+NodeBase.OnLoaded(FlaxEditor.Surface.SurfaceNodeActions)
   at FlaxEditor.Surface.Archetypes.BehaviorTree+Node.OnLoaded(FlaxEditor.Surface.SurfaceNodeActions)
   at FlaxEditor.Surface.VisjectSurfaceContext.OnControlLoaded(FlaxEditor.Surface.SurfaceControl, FlaxEditor.Surface.SurfaceNodeActions)
   at FlaxEditor.Surface.VisjectSurfaceContext.SpawnNode(FlaxEditor.Surface.GroupArchetype, FlaxEditor.Surface.NodeArchetype, FlaxEngine.Float2, System.Object[], System.Action`1<FlaxEditor.Surface.SurfaceNode>)
   at FlaxEditor.Surface.VisjectSurface.OnPrimaryMenuButtonClick(FlaxEditor.Surface.ContextMenu.VisjectCMItem, FlaxEditor.Surface.Elements.Box)
   at FlaxEditor.Surface.ContextMenu.VisjectCM.OnClickItem(FlaxEditor.Surface.ContextMenu.VisjectCMItem)
   at FlaxEditor.Surface.ContextMenu.VisjectCMItem.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.DropPanel.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEditor.GUI.ContextMenu.ContextMenuBase.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.GUI.WindowRootControl.OnMouseUp(FlaxEngine.Float2, FlaxEngine.MouseButton)
   at FlaxEngine.Window.Internal_OnMouseUp(FlaxEngine.Float2 ByRef, FlaxEngine.MouseButton)
   at FlaxEngine.Interop.NativeInterop+Invoker+InvokerNoRet2`3[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[FlaxEngine.Float2, FlaxEngine.CSharp, Version=1.7.6400.0, Culture=neutral, PublicKeyToken=null],[FlaxEngine.MouseButton, FlaxEngine.CSharp, Version=1.7.6400.0, Culture=neutral, PublicKeyToken=null]].MarshalAndInvoke(System.Object, FlaxEngine.Interop.ManagedHandle, IntPtr)
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(FlaxEngine.Interop.ManagedHandle, FlaxEngine.Interop.ManagedHandle, IntPtr, IntPtr)
```

**Comment by MineBill** [comment]
After some further digging, it looks like calling `InitState` on the managed code also causes an exception with a weird pattern: If i launch the editor with a debugger attached and press play, no exception is thrown and the tree appears to execute correctly. Then, when i exit play mode and enter again, an exception will be thrown while calling `InitState`. Without a debugger, an exception is always thrown. It looks like there is some underlying issue when calling managed code with behavior trees but that's out of my league, so i can only help so much.

**Comment by mafiesto4** [comment]
Fixed in 18aafb6fc7ef80f3b0f7931bd84d1083da12915a.
